### ISSUE TITLE
builtins.AttributeError: '_CaseInsensitiveDictionary' object has no attribute 'copy'

### ISSUE THREAD

**Issue by tonal** [twisted/txmongo#252]
versions:
txmongo            19.1.0
pymongo            3.9.0

Error in txmongo/connection.py line: 259 ConnectionPool.__init__
```
wc_options = self.__uri['options'].copy()
```
Forget `copy()` in `pymongo._CaseInsensitiveDictionary`

**Comment by IlyaSkriblovsky** [comment]
Thanks for reporting. PyMongo <3.9 used plain dict there...

**Comment by userlond** [comment]
Temporary fix is to downgrade to pymongo 3.8.0

**Comment by tonal** [comment]
@userlond I use monkeypath:
```python
import pymongo
if pymongo.version_tuple == (3, 9, 0):
  def _cid_copy(self):
    return self.__class__(**self)
  pymongo.common._CaseInsensitiveDictionary.copy = _cid_copy
```
